cluster-autoscaler/cloudprovider/aws/ec2_instance_types.go

```diff
@@ -111,6 +111,48 @@ var InstanceTypes = map[string]*instanceType{
 		MemoryMb:     7680,
 		GPU:          0,
 	},
+	"c5": {
+		InstanceType: "c5",
+		VCPU:         72,
+		MemoryMb:     0,
```

Comment:
No memory? That seems wrong

Reply:
Indeed this seems wrong but there are similar entries [elsewhere](https://github.com/kubernetes/autoscaler/pull/483/files#diff-e4d58ad5096a877c286c642a21646263R78).
I wonder if how they are used in awsprovider.

Reply:
Update: As far as I can see from code, if the `DescribeLaunchConfigurations` API returned an incomplete `instanceType` like `c5` in the launch configuration, this entry could be used.

However, as there's no such EC2 instance type, containing entries like "c5" doesn't break CA. Also, I believe you can safely remove entries like this.

---

The instance type like "c5" is actually meant for [EC2 Dedicated Hosts](https://aws.amazon.com/ec2/dedicated-hosts/pricing/?nc1=h_ls).

You can also confirm that by seeing a raw entry for "c5" which looks like:

```
$ cat ~/Downloads/index.json | jq '.products[] | select(.attributes.instanceType =="c5")'
{
  "sku": "EBPZ7HKS449M43K4",
  "productFamily": "Dedicated Host",
  "attributes": {
    "servicecode": "AmazonEC2",
    "location": "US East (N. Virginia)",
    "locationType": "AWS Region",
    "instanceType": "c5",
    "currentGeneration": "Yes",
    "instanceFamily": "Compute optimized",
    "vcpu": "72",
    "physicalProcessor": "Intel Xeon Platinum 8124M",
    "clockSpeed": "3.0 Ghz",
    "memory": "NA",
    "storage": "NA",
    "networkPerformance": "NA",
    "processorArchitecture": "64-bit",
    "tenancy": "Reserved",
    "operatingSystem": "Linux",
    "licenseModel": "No License required",
    "usagetype": "ReservedHostUsage:c5",
    "operation": "RunInstances",
    "ecu": "NA",
    "instanceCapacity18xlarge": "1",
    "instanceCapacity2xlarge": "8",
    "instanceCapacity4xlarge": "4",
    "instanceCapacity9xlarge": "2",
    "instanceCapacityLarge": "36",
    "instanceCapacityXlarge": "18",
    "normalizationSizeFactor": "NA",
    "physicalCores": "36",
    "preInstalledSw": "NA",
    "processorFeatures": "Intel AVX, Intel AVX2, Intel AVX512, Intel Turbo",
    "servicename": "Amazon Elastic Compute Cloud"
  }
}
```

whereas a "regular" instance type like c5.xlarge looks like:

```
$ cat ~/Downloads/index.json | jq '.products[] | select(.attributes.instanceType =="c5.xlarge")'
{
  "sku": "F3HZNTTFXFDM68NR",
  "productFamily": "Compute Instance",
  "attributes": {
    "servicecode": "AmazonEC2",
    "location": "US East (N. Virginia)",
    "locationType": "AWS Region",
    "instanceType": "c5.xlarge",
    "currentGeneration": "Yes",
    "instanceFamily": "Compute optimized",
    "vcpu": "4",
    "physicalProcessor": "Intel Xeon Platinum 8124M",
    "clockSpeed": "3.0 Ghz",
    "memory": "8 GiB",
    "storage": "EBS only",
    "networkPerformance": "Up to 10 Gigabit",
    "processorArchitecture": "64-bit",
    "tenancy": "Dedicated",
    "operatingSystem": "SUSE",
    "licenseModel": "No License required",
    "usagetype": "DedicatedUsage:c5.xlarge",
    "operation": "RunInstances:000g",
    "dedicatedEbsThroughput": "Upto 2250 Mbps",
    "ecu": "16",
    "enhancedNetworkingSupported": "Yes",
    "normalizationSizeFactor": "8",
    "preInstalledSw": "NA",
    "processorFeatures": "Intel AVX, Intel AVX2, Intel AVX512, Intel Turbo",
    "servicename": "Amazon Elastic Compute Cloud"
  }
}
```

So, probably we can filter those entries with `"productFamily": "Compute Instance"` so that only regular instance types are included in the generated `ec2_instance_types.go`.
It would be an another issue unrelated to this one though.

Reply:
Yeah, I didn't bother filtering those because they shouldn't affect CA anyway.